web/src/app/shared/components/upgrades/master/master.component.ts

```diff
@@ -0,0 +1,89 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import { Component, ElementRef, EventEmitter, OnInit, ViewChild } from '@angular/core';
+import { DynamicComponent, DynamicEvent } from '@app/shared/directives/dynamic/dynamic.directive';
+import { BaseDirective } from '@app/shared/directives';
+import { IMasterData, IValue, MasterService, whatShow } from './master.service';
+import { IUpgrade } from "@app/shared/components";
+import { UpgradeParameters } from "@app/shared/components/upgrades/upgrade.directive";
+
+@Component({
+  selector: 'app-master',
```

Comment:
What are differents with `web/src/app/shared/components/actions/master/master.component.ts` ?
And are identical selectors the norm?

Reply:
Component is used only in parent directories, so it won't be a problem.